[PTX-2813] Added Cassandra+CassandraStress, MongoDB+YCSB, ElasticSearch+Rally apps.

vinayakshnd opened the pull request.

thiguetta reviewed (approved):
LGTM, have you run torpedo-dev-test?

vinayakshnd commented:
If cassandra-stress faces any issue, they move to `crashLoopBackOff` state. For e.g in below case, Cassandra app pods have failed to spinned up. In that case, Cassandra-stress pods have failed.

```
kgpo
NAME                                READY   STATUS             RESTARTS   AGE
cassandra-0                         0/1     Pending            0          33h
cassandra-stress-5d6cd5ccd5-87ch5   0/1     CrashLoopBackOff   445        33h
```

Cassandra-stress logs:
```
******************** Stress Settings ********************
Command:
  Type: write
  Count: 1,000,000
  No Warmup: true
  Consistency Level: ONE
  Target Uncertainty: not applicable
  Key Size (bytes): 10
  Counter Increment Distibution: add=fixed(1)
Rate:
  Auto: true
  Min Threads: 4
  Max Threads: 1000
Population:
  Sequence: 1..1000000
  Order: ARBITRARY
  Wrap: true
Insert:
  Revisits: Uniform:  min=1,max=1000000
  Visits: Fixed:  key=1
  Row Population Ratio: Ratio: divisor=1.000000;delegate=Fixed:  key=1
  Batch Type: not batching
Columns:
  Max Columns Per Key: 5
  Column Names: [C0, C1, C2, C3, C4]
  Comparator: AsciiType
  Timestamp: null
  Variable Column Count: false
  Slice: false
  Size Distribution: Fixed:  key=64
  Count Distribution: Fixed:  key=5
Errors:
  Ignore: true
  Tries: 10
Log:
  No Summary: false
  No Settings: false
  File: /cassandra-stress.load.data
  Interval Millis: 1000
  Level: NORMAL
Mode:
  API: JAVA_DRIVER_NATIVE
  Connection Style: CQL_PREPARED
  CQL Version: CQL3
  Protocol Version: V4
  Username: null
  Password: null
  Auth Provide Class: null
  Max Pending Per Connection: 128
  Connections Per Host: 8
  Compression: NONE
Node:
  Nodes: [cassandra]
  Is White List: false
  Datacenter: null
Schema:
  Keyspace: keyspace1
  Replication Strategy: org.apache.cassandra.locator.SimpleStrategy
  Replication Strategy Pptions: {replication_factor=1}
  Table Compression: null
  Table Compaction Strategy: null
  Table Compaction Strategy Options: {}
Transport:
  factory=org.apache.cassandra.thrift.TFramedTransportFactory; truststore=null; truststore-password=null; keystore=null; keystore-password=null; ssl-protocol=TLS; ssl-alg=SunX509; store-type=JKS; ssl-ciphers=TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA;
Port:
  Native Port: 9042
  Thrift Port: 9160
  JMX Port: 7199
Send To Daemon:
  *not set*
Graph:
  File: null
  Revision: unknown
  Title: null
  Operation: WRITE
TokenRange:
  Wrap: false
  Split Factor: 1

java.lang.RuntimeException: java.lang.IllegalArgumentException: Failed to add contact point: cassandra
	at org.apache.cassandra.stress.settings.StressSettings.getJavaDriverClient(StressSettings.java:220)
	at org.apache.cassandra.stress.settings.SettingsSchema.createKeySpacesNative(SettingsSchema.java:79)
	at org.apache.cassandra.stress.settings.SettingsSchema.createKeySpaces(SettingsSchema.java:69)
	at org.apache.cassandra.stress.settings.StressSettings.maybeCreateKeyspaces(StressSettings.java:228)
	at org.apache.cassandra.stress.StressAction.run(StressAction.java:58)
	at org.apache.cassandra.stress.Stress.run(Stress.java:143)
	at org.apache.cassandra.stress.Stress.main(Stress.java:62)
Caused by: java.lang.IllegalArgumentException: Failed to add contact point: cassandra
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:925)
	at org.apache.cassandra.stress.util.JavaDriverClient.connect(JavaDriverClient.java:130)
	at org.apache.cassandra.stress.settings.StressSettings.getJavaDriverClient(StressSettings.java:211)
	... 6 more
Caused by: java.net.UnknownHostException: cassandra: Name or service not known
	at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:922)
	... 8 more
```

thiguetta commented:
@vinayakshnd is this PR still needed?
